tests/checks/set.fish

```diff
@@ -725,3 +725,68 @@ begin
     echo $CDPATH
     # CHECK: . /usr
 end
+
+# Function scope:
+set -f actuallyglobal "this one is global"
+set -qg actuallyglobal
+and echo "Yep, it's global"
+# CHECK: Yep, it's global
+set -S actuallyglobal
+#CHECK: $actuallyglobal: set in global scope, unexported, with 1 elements
+#CHECK: $actuallyglobal[1]: |this one is global|
+
+# Blocks aren't functions, "function" scope is still global:
+begin
+    set -f stillglobal "as global as the moon is wet"
+    echo $stillglobal
+    # CHECK: as global as the moon is wet
+end
+set -S stillglobal
+#CHECK: $stillglobal: set in global scope, unexported, with 1 elements
+#CHECK: $stillglobal[1]: |as global as the moon is wet|
+
```

Comment:
we could add a test for the case when a global of the same name is already defined

```fish
function test-function-scope
    set -f stillglobal 'setting variable in function scope will not clobber global variable'
end
test-function-scope
echo $stillglobal
#CHECK: as global as the moon is wet
```